packages/@aws-cdk/cdk/test/test.app.ts

```diff
@@ -1,120 +1,94 @@
 import cxapi = require('@aws-cdk/cx-api');
+import fs = require('fs');
 import { Test } from 'nodeunit';
+import os = require('os');
+import path = require('path');
 import { Construct, Resource, Stack, StackProps } from '../lib';
 import { App } from '../lib/app';
 
-//
-// this is the idiomatic way we want our apps to look like:
+function withApp(context: { [key: string]: any } | undefined, block: (app: App) => void) {
+  const outdir = fs.mkdtempSync(path.join(os.tmpdir(), 'cdk-app-test'));
+  process.env[cxapi.OUTDIR_ENV] = outdir;
 
-function main(...argv: string[]): any {
-  const app = new App([ 'myprog', ...argv ]);
-
-  // stacks are children of the program
-
-  const stack1 = new Stack(app, 'stack1', { env: { account: '12345', region: 'us-east-1' } });
-  new Resource(stack1, 's1c1', { type: 'DummyResource', properties: { Prop1: 'Prop1' } });
-  const r2 = new Resource(stack1, 's1c2', { type: 'DummyResource', properties: { Foo: 123 } });
+  if (context) {
```

Comment:
`process.env[cxapi.CONTEXT_ENV] = JSON.stringify(context || {})`?

Reply:
I actually want to remove this environment variable altogether. I tried to assigned `undefined` but for some weird reason, it will assign an `"undefined"` string to it instead of remove it.